src/category_theory/limits/presheaf.lean

```diff
@@ -308,14 +308,32 @@ def unique_extension_along_yoneda (L : (Cᵒᵖ ⥤ Type u₁) ⥤ ℰ) (hL : yo
   L ≅ extend_along_yoneda A :=
 nat_iso_of_nat_iso_on_representables _ _ (hL ≪≫ (is_extension_along_yoneda _).symm)
 
+-- TODO: This needs some lemmas describing the produced adjoint.
 /--
 If `L` preserves colimits and `ℰ` has them, then it is a left adjoint. Note this is a (partial)
 converse to `left_adjoint_preserves_colimits`.
 -/
-def is_left_adjoint_of_preserves_colimits (L : (Cᵒᵖ ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
+def is_left_adjoint_of_preserves_colimits (L : (C ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
   is_left_adjoint L :=
-{ right := restricted_yoneda (yoneda ⋙ L),
-  adj := (yoneda_adjunction _).of_nat_iso_left
-              (unique_extension_along_yoneda _ L (iso.refl _)).symm }
+begin
+  let L' : (Cᵒᵖᵒᵖ ⥤ Type u₁) ⥤ ℰ := ((op_op_equivalence C).congr_left).functor ⋙ L,
+  let : preserves_colimits L',
+  { refine ⟨λ J 𝒥₁, _⟩,
+    constructor,
+    refine λ K, _,
+    apply_instance },
+  let : is_left_adjoint L',
+  { refine ⟨restricted_yoneda (yoneda ⋙ L'), _⟩,
+    resetI,
+    apply (yoneda_adjunction _).of_nat_iso_left _,
+    { apply_instance },
+    apply (unique_extension_along_yoneda _ L' (iso.refl _)).symm },
+  let : is_left_adjoint (((op_op_equivalence C).congr_left).inverse ⋙ L'),
+  { apply adjunction.left_adjoint_of_comp _ _,
+    apply_instance,
+    apply this },
+  resetI,
```

Comment:
I suppose you could use `letI` instead of `let`, and then you wouldn't need `resetI` here.

Reply:
Yup, removed a couple of others too, thanks!